### PR TITLE
Add DisableNativeToolsetInstalls check to tools.ps1/.sh

### PR DESCRIPTION
**Pull request by sunandabalu**
- Moving the DisableNativeToolsetInstalls check into tools.ps1/.sh from build.ps1/.sh
- Update post-build SDL scripts  to set the "DisableNativeToolsetInstalls" variable. 

Test Build - https://dev.azure.com/dnceng/internal/_build/results?buildId=390365&view=results

### CODE REVIEW

File: eng/common/tools.ps1 [hunk 1]
```diff
@@ -418,7 +418,7 @@ function GetSdkTaskProject([string]$taskName) {
 }
 
 function InitializeNativeTools() {
-  if (Get-Member -InputObject $GlobalJson -Name "native-tools") {
+  if ((Get-Member -InputObject $GlobalJson -Name "native-tools") -and ($null -eq $env:DisableNativeToolsetInstalls)) {
```
Comment (chcosta) [inline eng/common/tools.ps1]: This condition is backwards, meaning, I'd expect it to check for the env variable first, and then (if that's null), to read the object for "native-tools".  Might just (for clarity), separate these out to something like...

```
function InitializeNativeTools() {
  if ($null -ne $env:DisableNativeToolsetInstalls) {
    return
  }
  …
```

File: eng/common/tools.ps1 [hunk 2]
```diff
@@ -418,7 +418,10 @@ function GetSdkTaskProject([string]$taskName) {
 }
 
 function InitializeNativeTools() {
-  if ((Get-Member -InputObject $GlobalJson -Name "native-tools") -and ($null -eq $env:DisableNativeToolsetInstalls)) {
+  if($null -eq $env:DisableNativeToolsetInstalls){
```
Comment (chcosta) [inline eng/common/tools.ps1]: nit: space between "if" and `(` and before `{`.

File: eng/common/tools.ps1 (hunk 1 above)
Comment (sunandabalu) [reply eng/common/tools.ps1]: Done

**Comment by sunandabalu** [comment]
> Do you have a link to the build which validates your changes fix the wpf build?

Test Build with the change for WinForms
https://dev.azure.com/dnceng/internal/_build/results?buildId=391874&view=results

File: eng/common/tools.ps1 [hunk 3]
```diff
@@ -605,7 +607,7 @@ Write-PipelineSetVariable -Name 'TMP' -Value $TempDir
 
 # Import custom tools configuration, if present in the repo.
 # Note: Import in global scope so that the script set top-level variables without qualification.
-if (!$env:DisableNativeToolsetInstalls) {
+if (!$PostBuildDisableNativeTools) {
```
Comment (chcosta) [inline eng/common/tools.ps1]: The condition name shouldn't be specific to the scenario you're trying to resolve, it should be specific to the action itself.  ie, something more general like "DisableConfigureToolsetImport"

File: eng/common/tools.ps1 [hunk 4]
```diff
@@ -49,6 +49,8 @@
 # An array of names of processes to stop on script exit if prepareMachine is true.
 $processesToStopOnExit = if (Test-Path variable:processesToStopOnExit) { $processesToStopOnExit } else { @("msbuild", "dotnet", "vbcscompiler") }
 
+$DisableConfigureToolsetImport = if (Test-Path variable:DisableConfigureToolsetImport) { $DisableConfigureToolsetImport } else { $null }
```
Comment (chcosta) [inline eng/common/tools.ps1]: Is this line needed?

Comment (sunandabalu) [reply eng/common/tools.ps1]: Yes it is. Just <variable_name>=$null is not defaulting to null if value not set. If you notice, all variables above are doing the same thing.

Comment (chcosta) [reply eng/common/tools.ps1]: The casing for this variable name is inconsistent, should be camel cased.

Comment (chcosta) [reply eng/common/tools.ps1]: If it's not defaulted to null, is there a code path where this throws an error, or is that ok?

Comment (sunandabalu) [reply eng/common/tools.ps1]: Without this, it was failing the other places where the codepath to include configure-toolset was being executed.

Comment (sunandabalu) [reply eng/common/tools.ps1]: > The casing for this variable name is inconsistent, should be camel cased.

Done

File: eng/common/tools.sh [hunk 5]
```diff
@@ -4,6 +4,7 @@
 
 # CI mode - set to true on CI server for PR validation build or official build.
 ci=${ci:-false}
+DisableConfigureToolsetImport=${DisableConfigureToolsetImport:-null}
```
Comment (chcosta) [inline eng/common/tools.sh]: Format for variable is inconsistent with the rest of the file.  Should be something like `disable_configure_toolset_import`

Comment (sunandabalu) [reply eng/common/tools.sh]: Not all seem to follow the pattern - https://github.com/dotnet/arcade/blob/master/eng/common/tools.sh#L50

Comment (chcosta) [reply eng/common/tools.sh]: That's true, and it's probably too late to fix.  Fortunately, it's not too late to update yours to align with the general conventions used in this file.

File: eng/common/tools.ps1 [hunk 6]
```diff
@@ -418,6 +420,9 @@ function GetSdkTaskProject([string]$taskName) {
 }
 
 function InitializeNativeTools() {
+  if ($env:DisableNativeToolsetInstalls) {
```
Comment (ViktorHofer) [inline eng/common/tools.ps1]: I believe this broke the disabling mechanism of the native toolsets. cc @chcosta, @jashook @trylek

Comment (ViktorHofer) [reply eng/common/tools.ps1]: It should be just `$DisableNativeToolsetInstalls` without the env.

Comment (chcosta) [reply eng/common/tools.ps1]: With strict mode on, it should probably be something like

` if (-Not( Test-Path variable:DisableNativetoolsetInstalls))`

@sunandabalu , can you take a look?

Comment (chcosta) [reply eng/common/tools.ps1]: Looks like Viktor already has a PR.  Thanks @ViktorHofer

File: eng/common/tools.sh [hunk 7]
```diff
@@ -252,6 +253,9 @@ function GetNuGetPackageCachePath {
 }
 
 function InitializeNativeTools() {
+  if [[ -z "${DisableNativeToolsetInstalls:-}" ]]; then
```
Comment (chcosta) [inline eng/common/tools.sh]: @sunandabalu , is this supposed to be `-n` instead of `-z`?

Comment (sunandabalu) [reply eng/common/tools.sh]: -z to check if its empty, if empty then exit.

Comment (chcosta) [reply eng/common/tools.sh]: Don't you want to exit if it's **not** empty?

Comment (chcosta) [reply eng/common/tools.sh]: For that matter, the fix viktor made seems off as well.  What's the intent of "disable" native toolset installs?  I interpret your implementation as, if this value is empty, then do nothing, if it is not defined, then skip native tools installs.  Something seems broken here, but perhaps the logic is getting jumbled in my head.

**Pull request merged**